lib/surro-gate/proxy.rb

```diff
@@ -111,16 +111,23 @@ def thread_stop
     end
 
     def reactor
-      # Atomically get an array of readable/writable monitors
-      monitors = @mutex.synchronize { @selector.select(0.1) || [] }
+      # Atomically get an array of readable monitors while also polling for writables
+      monitors = @mutex.synchronize do
+        @writer.select(0.1)
+        @reader.select(0.1) || []
+      end
       # Call each transmission proc and collect the results
-      callers = monitors.map { |m| m.value.call }
-      # Sleep for a short time if there was no transmission
-      sleep(0.1) if callers.none? && monitors.any?
+      monitors.map { |m| m.value.call }
```

Comment:
Also, happy to see this helps reduce the number of times the `sleep(0.1)` is necessary to almost nothing, and the true nature of why it was included in the implementation was discovered.  I was concerned that would be a bottle neck since a lot of data would most likely be flowing through here, and adding 100ms of a sleep is a lot if it happens to be called a bunch.